src/ai/backend/manager/api/resource.py

```diff
@@ -311,12 +314,23 @@ async def get_container_stats_for_period(request: web.Request, start_date, end_d
         result = await conn.execute(query)
         rows = result.fetchall()
 
+    async def _pipe_builder(r: Redis) -> RedisPipeline:
+        pipe = r.pipeline()
+        for row in rows:
+            pipe.get(str(row['id']))
+        return await pipe.execute()
```

Comment:
Just return the pipe using a synchronous function. ([ref](https://github.com/lablup/backend.ai-common/blob/main/tests/redis/test_pipeline.py))